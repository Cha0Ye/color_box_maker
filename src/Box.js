import React, { Component } from 'react';

class Box extends Component {

    render() {
        const { width, height, backgroundColor } = this.props.boxStyle;
        return (
            <div style={{ width, height, backgroundColor }}>
            <button onClick={ this.props.triggerDelete }>X</button>  
            </div>
        );
    }
}

export default Box;