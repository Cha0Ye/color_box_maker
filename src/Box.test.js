import React from 'react';
import Box from './Box';
import { shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

let wrapper;

beforeEach(()=>{
    const style = {width: '10px',
                    height: '10px', backgroundColor: 'blue'};
    wrapper = mount(<Box boxStyle={ style } />);
})

// smoke test
it('renders without crashing', () => {
    const style = {width: '10px',
                    height: '10px', backgroundColor: 'blue'};
  const wrapper = shallow(<Box boxStyle={ style } />);
});

// snapshot test
it('matches snapshot', () => {

    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});


