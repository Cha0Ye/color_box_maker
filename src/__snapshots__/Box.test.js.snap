// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Box
  boxStyle={
    Object {
      "backgroundColor": "blue",
      "height": "10px",
      "width": "10px",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "blue",
        "height": "10px",
        "width": "10px",
      }
    }
  >
    <button>
      X
    </button>
  </div>
</Box>
`;
